baseapp_core/rest_framework/serializers.py

```diff
@@ -21,6 +24,11 @@ def build_url_field(self, field_name, model_class):
 
     def build_standard_field(self, field_name, model_field):
         field_class, field_kwargs = super().build_standard_field(field_name, model_field)
+        # If the model exposes a public_id (via PublicIdMixin) replace id/pk with public_id
+        if field_name in ("id", "pk") and issubclass(self.Meta.model, PublicIdMixin):
```

Comment:
Is it possible to check with [field.primary_key](https://docs.djangoproject.com/en/5.2/ref/models/fields/#django.db.models.Field.primary_key) instead of checking by the `field_name`?  Because if I have a custom primary key name it will not work with this current implementation, you could check with `model_field.primary_key` or `field_class.primary_key`, can we try that?